### PR TITLE
Feature/improve desktop ui

### PR DESCRIPTION
**Pull request by spiroskous**
- [x] Check test coverage for each component.
- [x] Check checkstyle/linting for each component.
- [x] Check out code locally and review new classes & functions.
- [x] Run locally and test against the acceptance criteria.
- [x] Run Android E2E tests locally by doing 
```
cd android_client/e2e
invoke test
publish_test_artifacts
```

### CODE REVIEW

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java [hunk 1]
```diff
@@ -43,43 +44,78 @@ public Scene getScene() {
...
+
+        return scene;
     }
 
     private MigPane buildHeader() {
         MigPane header = new MigPane();
         header.setLayout("fill");
-        header.setStyle("-fx-background-color: #32292F");
-
+        header.setStyle("-fx-background-color:  #3C4F76;");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java]: Move this into the stylesheet (`resources/chitchat.css`). You can do
```
#base-header {
  -fx-background-color:  #3C4F76;
}
```

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java [hunk 2]
```diff
@@ -43,43 +44,78 @@ public Scene getScene() {
...
         Label title = new Label("Chit Chat");
-        title.setTextFill(Color.WHITE);
+        title.setWrapText(true);
+        title.setId("base-header-title");
+        if (this.baseStage.getConfiguration() != null
+                && this.baseStage.getConfiguration().getLoggedInUser() != null) {
+
+            logout = new Button("Log out");
+            logout.setOnAction(this);
+            logout.setStyle("-fx-background-color: transparent");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java [hunk 3]
```diff
@@ -43,43 +44,78 @@ public Scene getScene() {
...
+            logout.setId("logoutbtn");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java]: Should follow the existing form (*view*-*name*-*control type*) to be `base-logout-btn`.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java [hunk 4]
```diff
@@ -43,43 +44,78 @@ public Scene getScene() {
...
         return header;
     }
 
     private MigPane buildFooter() {
         MigPane footer = new MigPane();
+        footer.setId("base-footer");
         footer.setLayout("fill");
-        footer.setStyle("-fx-background-color: #705D56;");
+
+        footer.setStyle("-fx-background-color:  #3C4F76;");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java [hunk 5]
```diff
@@ -43,43 +44,78 @@ public Scene getScene() {
...
 
         Label credits = new Label("Created by: Monarchs of Coding");
-        credits.setId("credits");
         credits.setTextFill(Color.WHITE);
-        footer.add(credits);
-
+        credits.setId("credits");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/BaseView.java]: Should be `base-footer-credits`

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/LoginView.java [hunk 6]
```diff
@@ -41,62 +47,64 @@
...
         this.passwordField.setOnAction(this);
-        loginForm.add(this.passwordField,"grow,wrap");
+        this.passwordField.setMinWidth(295.0); // 295.0 is the magic width
+        this.loginForm.add(this.passwordField, "grow,wrap");
 
-        this.loginBtn = new Button("Login");
+        this.loginBtn = new JFXButton("Login");
         this.loginBtn.setOnAction(this);
         this.loginBtn.setId("login-login-btn");
+        this.loginBtn.setStyle("-fx-background-color: rgba(93,200,127,0.99)");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/LoginView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/LoginView.java [hunk 7]
```diff
@@ -41,62 +47,64 @@
...
+        this.loginBtn.setMinWidth(295.0); // 295.0 is the magic width
         loginForm.add(this.loginBtn, "wrap, grow");
 
-        this.registerBtn = new Button("Register");
+        this.registerBtn = new JFXButton("Register");
         this.registerBtn.setId("login-register-btn");
         this.registerBtn.setOnAction(this);
-        loginForm.add(this.registerBtn, "wrap,grow");
+        this.registerBtn.setStyle("-fx-background-color: rgba(129,178,248,0.99)");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/LoginView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/LoginView.java [hunk 8]
```diff
@@ -41,62 +47,64 @@
...
+        this.registerBtn.setMinWidth(295.0); // 295.0 is the magic width
+        this.loginForm.add(this.registerBtn, "wrap,grow");
 
         this.unexpectedErrors = new Label();
         this.unexpectedErrors.setId("login-errors-lbl");
         this.unexpectedErrors.setTextFill(Color.RED);
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/LoginView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/RegistrationView.java [hunk 9]
```diff
@@ -86,9 +92,11 @@ public MigPane getContentPane() {
         this.passwordCheckErrors.setVisible(false);
         registerForm.add(this.passwordCheckErrors, "span");
 
-        this.registerBtn = new Button("Register");
+        this.registerBtn = new JFXButton("Register");
         this.registerBtn.setOnAction(this);
         this.registerBtn.setId("register-register-btn");
+        this.registerBtn.setStyle("-fx-background-color: rgba(93,200,127,0.99)");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/RegistrationView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/RegistrationView.java [hunk 10]
```diff
@@ -97,9 +105,11 @@ public MigPane getContentPane() {
         this.unexpectedErrors.setVisible(false);
         registerForm.add(this.unexpectedErrors, "span");
 
-        this.loginBtn = new Button("Login");
+        this.loginBtn = new JFXButton("Login");
         this.loginBtn.setOnAction(this);
         this.loginBtn.setId("register-login-btn");
+        this.loginBtn.setStyle("-fx-background-color: rgba(129,178,248,0.99)");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/authentication/RegistrationView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java [hunk 11]
```diff
@@ -82,34 +87,34 @@ void showConversation(Conversation c) {
         }
 
         Label header = new Label(String.format("Chat with: %s", c.getOtherParticipant().getUsername()));
+        header.setFont(Font.font(null, FontWeight.BOLD, 15));
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java [hunk 12]
```diff
@@ -82,34 +87,34 @@ void showConversation(Conversation c) {
...
+        header.setTextFill(Color.BLUE);
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java]: Use the stylesheet

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java [hunk 13]
```diff
@@ -82,34 +87,34 @@ void showConversation(Conversation c) {
...
         this.errormessage.setTextFill(Color.RED);
         this.errormessage.setId("conversation-error-lbl");
         this.errormessage.setVisible(false);
-        this.conversationPane.add(errormessage, "span,grow,wrap");
+        this.conversationPane.add(errormessage, "span, growx");
 
-        this.sendbtn = new Button("Send");
+        this.sendbtn = new JFXButton("Send");
         this.sendbtn.setId("conversation-send-btn");
+        this.sendbtn.setStyle("-fx-background-color: rgba(32,137,72,0.99)");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/SearchView.java [hunk 14]
```diff
@@ -80,37 +84,39 @@ public MigPane getContentPane() {
...
-        searchForm.add(this.usernameField, "span, grow");
+        searchForm.add(this.errorUserMessage,"span");
+        searchForm.add(this.usernameField, "span");
 
-        this.searchBtn = new Button("Search");
+        this.searchBtn = new JFXButton("Search");
         this.searchBtn.setId("search-btn");
         this.searchBtn.setOnAction(this);
-        searchForm.add(this.searchBtn, "span, grow");
+        this.searchBtn.setStyle("-fx-background-color: rgba(56,64,166,0.87)");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/SearchView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/SearchView.java [hunk 15]
```diff
@@ -80,37 +84,39 @@ public MigPane getContentPane() {
...
+        this.searchList.setItems(this.observableUserList);
         this.searchList.setId("search-user-list");
         searchForm.add(this.searchList, "span, grow");
 
-        this.startConversationBtn = new Button("Start Chat");
+        this.startConversationBtn = new JFXButton("Start Chat");
         this.startConversationBtn.setId("search-chat-btn");
         this.startConversationBtn.setOnAction(this);
-        searchForm.add(this.startConversationBtn, "span, wrap");
+        this.startConversationBtn.setStyle("-fx-background-color: rgba(32,137,72,0.99)");
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/SearchView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java [hunk 16]
```diff
@@ -82,34 +87,34 @@ void showConversation(Conversation c) {
...
-        this.newMessageField = new TextField();
+        this.newMessageField = new JFXTextField();
         this.newMessageField.setPromptText("Enter Message: ");
         this.newMessageField.setId("conversation-message-fld");
         this.newMessageField.setOnAction(this);
-        this.conversationPane.add(newMessageField, "span,grow");
+        this.conversationPane.add(newMessageField, "span, growx");
 
         this.errormessage = new Label();
         this.errormessage.setTextFill(Color.RED);
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/ConversationView.java]: Use the stylesheet.

File: java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/SearchView.java [hunk 17]
```diff
@@ -80,37 +84,39 @@ public MigPane getContentPane() {
...
+        searchForm.add(this.startConversationBtn, "span");
 
         this.errorMessage = new Label();
         this.errorMessage.setId("search-error-messages");
         this.errorMessage.setVisible(false);
         this.errorMessage.setTextFill(Color.RED);
```
Comment (VJftw) [inline java_client/ChitChatDesktop/src/com/moc/chitchat/view/main/SearchView.java]: Use the stylesheet.

**Review by VJftw — CHANGES_REQUESTED** [review]

**Review by AigerimSerikbekova — APPROVED** [review]

**Comment by VJftw** [comment]
Ran the android_client/e2e locally. All passing :100:

**Review by VJftw — APPROVED** [review]

**Review by Ozhan-Azizi — CHANGES_REQUESTED** [review]
can you try to make the listview bigger cause it is a line and we miss the returned user after usersearch

**Review by AigerimSerikbekova — APPROVED** [review]

**Review by Ozhan-Azizi — APPROVED** [review]

**Pull request merged**